src/PhotometryModel.cc

```diff
@@ -0,0 +1,38 @@
+#include "lsst/log/Log.h"
+
+#include "lsst/jointcal/CcdImage.h"
+#include "lsst/jointcal/PhotometryModel.h"
+
+namespace lsst {
+namespace jointcal {
+
+bool PhotometryModel::validate(CcdImageList const &ccdImageList) const {
+    bool check = true;
+    for (auto const &ccdImage : ccdImageList) {
+        check &= checkPositiveOnBBox(*ccdImage);
+    }
+    return check;
+}
+
+bool PhotometryModel::checkPositiveOnBBox(CcdImage const &ccdImage) const {
+    bool check = true;
+    double result;
+    auto bbox = ccdImage.getImageFrame();
+    std::vector<double> xRange = {bbox.xMin, bbox.getCenter().x, bbox.xMax};
+    std::vector<double> yRange = {bbox.yMin, bbox.getCenter().y, bbox.yMax};
+    for (auto const &x : xRange)
+        for (auto const &y : yRange) {
```

Comment:
Why not put the initializer lists in the loop definition (i.e., get rid of `xRange` and `yRange` entirely)? I think it would improve readability.

Reply:
Because I didn't even think of it. Good idea.